ml-agents/mlagents/trainers/tf_policy.py

```diff
@@ -194,6 +200,34 @@ def remove_memories(self, agent_ids):
             if agent_id in self.memory_dict:
                 self.memory_dict.pop(agent_id)
 
+    def make_empty_previous_action(self, num_agents):
```

Comment:
Is there any difference between how we handle memories and previous actions besides the the dtype? Worth combining into a small utility class?

Reply:
I think there is going to be a bigger refactor of this part coming later. I think it is fine for now. What do you think @ervteng ?

Reply:
Might be worth combining - but then again, there will only be two things we store between steps. Also, the "when" they're added is different - memories can be added after the get_action, previous_actions must be added before. 

I think there is no way around it, this logic will still need to live in the Policy for Python inference to work. The only other place I can think of is the env_manager, but that would be messy.